src/components/combo/combo.ts

```diff
@@ -0,0 +1,654 @@
+import IgcIconComponent from '../icon/icon.js';
+import { NavigationController } from './controllers/navigation.js';
+import { SelectionController } from './controllers/selection.js';
+import { IgcToggleController } from '../toggle/toggle.controller.js';
+import { DataController } from './controllers/data.js';
+import { IgcToggleComponent } from '../toggle/types.js';
+import {
+  Keys,
+  Values,
+  ComboRecord,
+  GroupingDirection,
+  FilteringOptions,
+  IgcComboEventMap,
+} from './types.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { partNameMap } from '../common/util.js';
+import { filteringOptionsConverter } from './utils/converters.js';
+import { EventEmitterMixin } from '../common/mixins/event-emitter.js';
+import { Constructor } from '../common/mixins/constructor.js';
+
+defineComponents(
+  IgcIconComponent,
+  IgcComboListComponent,
+  IgcComboItemComponent,
+  IgcComboHeaderComponent,
+  IgcInputComponent
+);
+
+/**
+ * @element igc-combo
+ *
+ * @slot prefix - Renders content before the input.
+ * @slot suffix - Renders content after input.
+ * @slot header - Renders a container before the list of options.
+ * @slot footer - Renders a container after the list of options.
+ * @slot helper-text - Renders content below the input.
+ * @slot toggle-icon - Renders content inside the suffix container.
+ * @slot clear-icon - Renders content inside the suffix container.
+ *
+ * @fires igcFocus - Emitted when the select gains focus.
+ * @fires igcBlur - Emitted when the select loses focus.
+ * @fires igcChange - Emitted when the control's selection has changed.
+ * @fires igcOpening - Emitted just before the list of options is opened.
+ * @fires igcOpened - Emitted after the list of options is opened.
+ * @fires igcClosing - Emitter just before the list of options is closed.
+ * @fires igcClosed - Emitted after the list of options is closed.
+ *
+ * @csspart label - The encapsulated text label.
+ * @csspart input - The main input field.
+ * @csspart prefix - The prefix wrapper.
+ * @csspart suffix - The suffix wrapper.
+ * @csspart toggle-icon - The toggle icon wrapper.
+ * @csspart clear-icon - The clear icon wrapper.
+ * @csspart helper-text - The helper text wrapper.
+ * @csspart search-input - The search input field.
+ * @csspart list-wrapper - The list of options wrapper.
+ * @csspart list - The list of options box.
+ * @csspart item - Represents each item in the list of options.
+ * @csspart group-header - Represents each header in the list of options.
+ * @csspart active - Appended to the item parts list when the item is active.
+ * @csspart selected - Appended to the item parts list when the item is selected.
+ * @csspart checkbox - Represents each checkbox of each list item.
+ * @csspart checkbox-indicator - Represents the checkbox indicator of each list item.
+ * @csspart checked - Appended to checkbox parts list when checkbox is checked.
+ * @csspart empty - The container holding the empty content.
+ */
```

Comment:
I guess the Blazor decorator for inter-component dependencies should be added here

Reply:
Added.